test/metabase/query_processor_test/parameters_test.clj

```diff
@@ -149,3 +149,40 @@
             results (qp/process-query query)]
         (is (= [[1]]
                (mt/formatted-rows [int] results)))))))
+
+(deftest native-with-spliced-params-test
+  (testing "Make sure we can convert a parameterized query to a native query with spliced params"
+    (testing "Multiple values"
+      (mt/dataset airports
+        (is (= {:query  "SELECT NAME FROM COUNTRY WHERE \"PUBLIC\".\"COUNTRY\".\"NAME\" IN ('US', 'MX')"
+                :params nil}
+               (qp/query->native-with-spliced-params
+                {:type       :native
+                 :native     {:query         "SELECT NAME FROM COUNTRY WHERE {{country}}"
+                              :template-tags {"country"
+                                              {:name         "country"
+                                               :display-name "Country"
+                                               :type         :dimension
+                                               :dimension    [:field-id (mt/id :country :name)]
+                                               :widget-type  :category}}}
+                 :database   (mt/id)
+                 :parameters [{:type   :location/country
+                               :target [:dimension [:template-tag "country"]]
+                               :value  ["US" "MX"]}]})))))
+
+    (testing "Comma-separated numbers"
+      (is (= {:query  "SELECT * FROM VENUES WHERE \"PUBLIC\".\"VENUES\".\"PRICE\" IN (1, 2)"
+              :params []}
+             (qp/query->native-with-spliced-params
+              {:type       :native
+               :native     {:query         "SELECT * FROM VENUES WHERE {{price}}"
+                            :template-tags {"price"
+                                            {:name         "price"
+                                             :display-name "Price"
+                                             :type         :dimension
+                                             :dimension    [:field-id (mt/id :venues :price)]
+                                             :widget-type  :category}}}
+               :database   (mt/id)
+               :parameters [{:type   :category
+                             :target [:dimension [:template-tag "price"]]
+                             :value  [1 2]}]}))))))
```

Comment:
These ended up being unrelated but I wrote these tests anyway so it doesn't hurt to keep them